ui/app-loader/src/integrations.ts

```diff
@@ -363,9 +339,20 @@ export const handleExtPointMountOfApps = (
           navigateTo: navigateTo(props.integrationConfigs, logger),
           navigateToModal: navigateToModal,
           getAppRoutes: appName => props.integrationConfigs.get(appName).routes,
-          getMenuItems: () => state$.pipe(getStateSlice('menuItems')),
         },
       });
+
+      // each app must have menuItems exposed in config, widgets do not
+      if (config.menuItems) {
+        dispatchEvent(
```

Comment:
you can use uiEvents instead of custom event.
This way is also easier to keep track of the events